multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/SupportedParametersChecker.java

```diff
@@ -0,0 +1,114 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.cloudfoundry.multiapps.controller.core.model.SupportedParameters;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Hook;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.cloudfoundry.multiapps.mta.model.RequiredDependency;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+
+@Named("supportedParametersChecker")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class SupportedParametersChecker {
+
+    private List<String> unknownParameters;
+
+    @Inject
+    private ReferenceFinder finder;
+
+    public List<String> getUnknownParameters(DeploymentDescriptor descriptor) {
+        List<String> unsupportedAndNotReferencedParameters = new ArrayList<>();
+        for (String parameter : findUnsupportedParameters(descriptor)) {
+            if (!this.finder.isParameterReferenced(descriptor, parameter)) {
+                unsupportedAndNotReferencedParameters.add(parameter);
+            }
+        }
+        return unsupportedAndNotReferencedParameters;
+    }
+
+    private List<String> findUnsupportedParameters(DeploymentDescriptor descriptor) {
+        this.unknownParameters = new ArrayList<>();
+        descriptor.getModules()
+                  .forEach(this::checkModuleParameters);
+        descriptor.getResources()
+                  .forEach(this::checkResourceParameters);
+        checkParameters(getGlobalParameters(), descriptor.getParameters());
+        return unknownParameters;
+    }
+
+    private void checkModuleParameters(Module module) {
+        checkParameters(getModuleParameters(), module.getParameters());
+        getModuleHooks(module).forEach(this::checkHookParameters);
+        module.getRequiredDependencies()
+              .forEach(dependency -> checkParameters(getDependencyParameters(), dependency.getParameters()));
```

Comment:
invoking getDependencyParameters() twice here seems bad, just save it in a variable